lib/flowViewport/selection/fvpSelection.h

```diff
@@ -93,17 +93,10 @@ class Selection
     GetVectorDataSource(const PXR_NS::SdfPath& primPath) const;
 
 private:
-
-    struct _PrimSelectionState {
-        // Container data sources conforming to HdSelectionSchema
-        std::vector<PXR_NS::HdDataSourceBaseHandle> selectionSources;
-
-        PXR_NS::HdDataSourceBaseHandle GetVectorDataSource() const;
-    };
     
-    // Maps prim path to data sources to be returned by the vector data
+    // Maps prim path to selections to be returned by the vector data
     // source at locator selections.
-    std::map<PXR_NS::SdfPath, _PrimSelectionState> _pathToState;
+    std::map<PXR_NS::SdfPath, std::vector<PrimSelection>> _pathToSelections;
```

Comment:
Now only stores the `PrimSelection` objects directly, which are fairly small, and which each correspond to a selection data source, constructed in `GetVectorDataSource`.